src/main/java/org/broadinstitute/hellbender/tools/exome/cnlohcaller/CNLOHCaller.java

```diff
@@ -0,0 +1,803 @@
+                                   final double credibleLow, final double credibleHigh) {
+        ParamUtils.inRange(rho, 0.0, 1.0, "Invalid rho value: " + rho + ".  Must be [0,1]");
+        ParamUtils.isPositiveOrZero(m, "M must be positive.");
+        ParamUtils.isPositiveOrZero(n, "N must be positive.");
+
+        if (Double.isNaN(credibleMode) || Double.isNaN(credibleLow) || Double.isNaN(credibleHigh)){
+            return 1;
+        }
+
+        final double maf = calculateMaf(rho, m, n);
+        return Math.max(calculateDoubleGaussian(maf, credibleMode, credibleLow, credibleHigh), MIN_L);
+
+    }
+
+    /**
+     * Calculates the minor allelic fraction for given rho, M, and N.
+     *
+     *  Output is truncated at MIN_L
+     *
+     *  @param rho CCF*purity
+     * @param m absolute copy number of allele1
+     * @param n absolute copy number of allele2
+     * @return calculated minor allele fraction
+     */
+    @VisibleForTesting
+    static double calculateMaf(final double rho, final int m, final int n) {
+
+        ParamUtils.inRange(rho, 0.0, 1.0, "Invalid rho value: " + rho + ".  Must be [0,1]");
+        ParamUtils.isPositiveOrZero(m, "M must be >= 0");
+        ParamUtils.isPositiveOrZero(n, "N must be >= 0");
+
+        double result = ((1-rho) + rho * (Math.min(n,m))) / ( (1-rho)*2 + rho * (n+m) );
+
+        // If result is NaN, that means that we had rho = 1 and M,N = 0
+        if ((result < MIN_L) || (Double.isNaN(result))) {
+            result = MIN_L;
+        }
+        return result;
+    }
+
+    /**
+     * This function takes two Gaussian distributions and uses one for the values below the mode and the other for values above the
+     *  mode.
+     *
+     *  If the mode is outside the credible interval, the calculation is done as if the interval boundary is
+     *   {@link CNLOHCaller#MIN_DIST_FROM_MODE} above/below the mode.
+     *
+     *
+     * @param val value to calculate the "pdf"
+     * @param credibleMode mode of the presumed distribution
+     * @param credibleLow 95% confidence interval on the low tail
+     * @param credibleHigh 95% confidence interval on the high tail
+     * @return pdf with a minimum value of {@link CNLOHCaller#MIN_L}
+     */
+    @VisibleForTesting
+    static double calculateDoubleGaussian(final double val, final double credibleMode, final double credibleLow,
+                                             final double credibleHigh) {
+
+        double hiDist = credibleHigh - credibleMode;
+        if (credibleMode >= credibleHigh) {
+            hiDist = MIN_DIST_FROM_MODE;
+        }
+        double loDist = credibleMode - credibleLow;
+        if (credibleMode <= credibleLow) {
+            loDist = MIN_DIST_FROM_MODE;
+        }
+
+        if (val >= credibleMode) {
+            return new NormalDistribution(credibleMode, hiDist / 1.96).density(val);
+        } else {
+            return new NormalDistribution(credibleMode, loDist / 1.96).density(val);
```

Comment:
Sheesh, you and those ternaries....

From the Apache Commons source code
```
...
In case no sampling is
needed for the created distribution, it is advised to pass {@code null}
as random generator via the appropriate constructors to avoid the
additional initialisation overhead.
```
Both Done...